src/framework/filterControl/forms/MultipleSelectFilterForm.jsx

```diff
@@ -19,16 +19,17 @@ export default class MultipleSelectFilterForm extends Component {
 
     this.onCheckBoxClick = this.onCheckBoxClick.bind(this);
     this.onClickAddButton = this.onClickAddButton.bind(this);
-    this.options = this.props.filterOption.comparisonParameters.oneOfOptions;
+    this.options = props.filterOption.comparisonParameters.oneOfOptions;
 
     let selectedOptions;
-
     if (this.props.comparisonValue === undefined) {
       selectedOptions = (new Array(this.options.length)).fill(false);
     } else {
-      selectedOptions = this.options.map(option => (
-        this.props.comparisonValue.indexOf(option) !== -1
-      ));
+      selectedOptions = this.options.map(option =>
+        Boolean(props.comparisonValue.find(
+          element => element.value === option.value)
+        )
+      );
```

Comment:
@ahollesen you are right. I would still prefer comparing it with `undefined`, but I don't mind as it is now.